floodlight/io/sportradar.py

```diff
@@ -0,0 +1,199 @@
+        team_name = home_away_link[competitor[0]][1] if len(competitor) == 1 else None
+        pID = event["player"]["id"] if "player" in event else None
+        player_name = event["player"]["name"] if pID else None
+
+        event_name = event["type"]
+
+        # extract time codes and matchclock
+        time_stamp = datetime.datetime.fromisoformat(event["time"])
+        time_delta = time_stamp - segment_start
+        gameclock = time_delta.seconds
+        if "match_clock" in event:
+            match_clock = event["match_clock"]
+            minute, second = [int(x) for x in match_clock.split(":")]
+        else:
+            minute, second = (None, None)
+
+        # extract optional event information
+        outcome = event["outcome"] if "outcome" in event else None
+        home_score = event["home_score"] if "home_score" in event else None
+        away_score = event["away_score"] if "away_score" in event else None
+        scorer = event["scorer"] if "scorer" in event else None
+        assists = event["assists"] if "assists" in event else None
+        zone = event["zone"] if "zone" in event else None
+        shot_type = event["shot_type"] if "shot_type" in event else None
+        players = event["players"] if "players" in event else None
+
+        # add event to team event list
+        for team in competitor:
+            team_event_lists[team][segment]["eID"].append(eID)
+            team_event_lists[team][segment]["gameclock"].append(gameclock)
+            team_event_lists[team][segment]["time_stamp"].append(time_stamp)
+            team_event_lists[team][segment]["minute"].append(minute)
+            team_event_lists[team][segment]["second"].append(second)
+            team_event_lists[team][segment]["pID"].append(pID)
+            team_event_lists[team][segment]["player_name"].append(player_name)
+            team_event_lists[team][segment]["tID"].append(tID)
+            team_event_lists[team][segment]["team_name"].append(team_name)
+            team_event_lists[team][segment]["mID"].append(mID)
+            team_event_lists[team][segment]["event_name"].append(event_name)
+            team_event_lists[team][segment]["home_score"].append(home_score)
+            team_event_lists[team][segment]["away_score"].append(away_score)
+            team_event_lists[team][segment]["scorer"].append(scorer)
+            team_event_lists[team][segment]["assists"].append(assists)
+            team_event_lists[team][segment]["zone"].append(zone)
+            team_event_lists[team][segment]["shot_type"].append(shot_type)
+            team_event_lists[team][segment]["outcome"].append(outcome)
+            team_event_lists[team][segment]["players"].append(players)
+
+        # flexible parser return for all segments and teams
+        data_objects = {
+            segment: {
+                team: Events(events=pd.DataFrame(data=team_event_lists[team][segment]))
```

Comment:
the Events class has an attribute "direction" that highlights the playing direction ("lr" or "rl"). if thats available in the data it would be nice to include. if not, bad luck

Reply:
Bad luck it is.

Reply:
unlucky